### PR TITLE
V2 base

### PR DESCRIPTION
**Pull request by cristovaoth**

### CODE REVIEW

File: packages/evm/contracts/Types.sol [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    Dynamic,
+    Tuple,
+    Array,
+    AbiEncoded
+}
+
+enum Operator {
+    // 00:    EMPTY EXPRESSION (default, always passes)
+    //          paramType: Static / Dynamic
+    //          🚫 children
```
Comment (jfschwarz) [inline packages/evm/contracts/Types.sol:22]:
```suggestion
    //          paramType: Static / Dynamic / Tuple / Array
    //          ❓ children (only for paramType: Tuple / Array to describe their structure)
```

File: packages/evm/contracts/Types.sol [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    /* 11: */ _ComplexPlaceholder11,
+    /* 12: */ _ComplexPlaceholder12,
+    /* 13: */ _ComplexPlaceholder13,
+    /* 14: */ _ComplexPlaceholder14,
+    /* 15: */ _ComplexPlaceholder15,
+    /* 16: */ _ComplexPlaceholder16,
+    // ------------------------------------------------------------
+    // 17-31: COMPARISON EXPRESSIONS
+    //          paramType: Static / Dynamic / Tuple / Array / AbiEncoded
+    //          🚫 children
```
Comment (jfschwarz) [inline packages/evm/contracts/Types.sol:54]:
```suggestion
    //           paramType: Static / Dynamic / Tuple / Array
    //          ❓ children (only for paramType: Tuple / Array to describe their structure)
```

(While technically possible to use EqualTo on AbiEncoded it does not make sense in practice. The param type can be declared as `Dynamic` and the comparison can be performed directly on the bytes instead.)

**Review by jfschwarz — COMMENTED** [review]
some clarifying edits on the Operator comments. (I've closed the [previous PR](https://github.com/gnosis/zodiac-modifier-roles/pull/193) addressing this)

**Review by samepant — APPROVED** [review]
🔢 🔥

**Pull request merged**